File: Programmers/Level 2/20.cpp
#include <string>
#include <iostream>
#include <stack>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    int sz = s.size();
    stack<char> st;

    for (int i = 0; i < sz; i++) {
        if (s[i] == '(')
            st.push(s[i]);
        else if (s[i] == ')') {
            if (st.empty()) {
                answer = false;
                break;
            }
            st.pop();
        }
    }

    if (!st.empty())
        answer = false;
    return answer;
}

File: Programmers/Level 2/20.py
def solution(s):    
    cnt = 0
    for i in s:
        if i == '(': # 여는 괄호
            cnt += 1
        else: # 닫는 괄호
            if cnt == 0:
                return False
            cnt -= 1
    if cnt == 0:
        return True
    else:
        return False
